client/telemetry/src/lib.rs

```diff
@@ -495,3 +559,26 @@ macro_rules! format_fields_to_json {
 		)*
 	}};
 }
+
+#[derive(Debug, Clone)]
+struct IdGenerator;
+
+impl IdGenerator {
+	fn new() -> Self {
+		Self
+	}
+
+	fn next(&mut self) -> Id {
+		TELEMETRY_ID_COUNTER.inc()
+	}
+}
+
+/// A trait used by the Substrate's Client to get a [`TelemetryHandle`] or to initialize the
+/// [`Telemetry`].
+pub trait ClientTelemetry {
```

Comment:
Why do we need this? The telemetry should be passed everywhere it is required and not fetched from the `Client`.

Reply:
Done in 30d11b53a3